src/results/bulk_write.rs

```diff
@@ -2,84 +2,180 @@
 
 use std::{collections::HashMap, fmt::Debug};
 
-use serde::Serialize;
-use serde_with::skip_serializing_none;
-
 use crate::{
+    error::bulk_write::PartialBulkWriteResult,
     results::{DeleteResult, InsertOneResult, UpdateResult},
-    serde_util::serialize_indexed_map,
 };
 
-#[skip_serializing_none]
-#[derive(Clone, Debug, Serialize)]
-#[serde(rename_all = "camelCase")]
+#[cfg_attr(test, serde_with::skip_serializing_none)]
+#[derive(Clone, Debug, Default)]
+#[cfg_attr(test, derive(serde::Serialize))]
+#[cfg_attr(test, serde(rename_all = "camelCase"))]
 #[non_exhaustive]
-pub struct BulkWriteResult {
+pub struct SummaryBulkWriteResult {
     pub inserted_count: i64,
+    pub matched_count: i64,
+    pub modified_count: i64,
     pub upserted_count: i64,
+    pub deleted_count: i64,
+}
+
+#[cfg_attr(test, serde_with::skip_serializing_none)]
+#[derive(Clone, Debug, Default)]
+#[cfg_attr(test, derive(serde::Serialize))]
+#[cfg_attr(test, serde(rename_all = "camelCase"))]
+#[non_exhaustive]
+pub struct VerboseBulkWriteResult {
+    pub inserted_count: i64,
```

Comment:
I wonder if it'd be worth having `pub summary: SummaryBulkWriteResult` instead of duplicating the fields?  API-wise it makes the superset relationship clear, and it would mean the impl would have a little less repeated code.

Reply:
yeah I think that makes sense. this might be the first time I've wanted subclasses in Rust...